pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -242,25 +244,64 @@ func (c *Controller) updateServiceNodePortAddresses(svcs ...*model.Service) bool
 		return false
 	}
 	for _, svc := range svcs {
+		if c.isNodePortGatewayService(svc) && svc.Attributes.ExternalTrafficPolicy == model.ExternalTrafficPolicyLocal {
+			endpoints := c.endpoints.buildIstioEndpointsWithService(svc.Attributes.Name, svc.Attributes.Namespace, svc.Hostname)
+			if features.EnableK8SServiceSelectWorkloadEntries {
+				fep := c.collectWorkloadInstanceEndpoints(svc)
+				endpoints = append(endpoints, fep...)
+			}
+			nodesWithEndpoints := getWorkloadNodeLocations(endpoints)
+			c.Lock()
+			c.serviceToWorkloadNodesMap[svc.Hostname] = nodesWithEndpoints
```

Comment:
I get to know serviceToWorkloadNodesMap stores the svc --> nodes (where the service instance resides) 

 Same performance concern to do it here.

Reply:
This cost is limited to node port gateways with external traffic policy set to Local. As I mentioned earlier, in practice, we don't expect a lot of them. So I think it should be fine. One potential concern could be the call on EDS update which is more frequent than service or node updates. However, this is applicable only for specific kind of Gateway services. AFAIK, it would not be too many. So it should be fine?